src/main/java/fi/tkgwf/ruuvi/utils/Utils.java

```diff
@@ -44,10 +57,11 @@ public static String hexToAscii(String hex) {
      * @return the MAC address, without spaces
      */
     public static String getMacFromLine(String line) {
-        if (StringUtils.isBlank(line)) {
+        String substringOfLine = line.substring(23);

```

Comment:
Taking the substring here is not nullsafe. You could move the substringing to the split-line. Or if you want to go fancy (optional), remove the substringing completely and instead rebuild the mac address at the end at an offset.